apis/v1alpha1/clientsettingspolicy_types.go

```diff
@@ -0,0 +1,149 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	gatewayv1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+)
+
+// +kubebuilder:object:root=true
+// +kubebuilder:storageversion
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:categories=nginx-gateway-fabric,shortName=cspolicy
+// +kubebuilder:printcolumn:name="Age",type=date,JSONPath=`.metadata.creationTimestamp`
+// +kubebuilder:metadata:labels="gateway.networking.k8s.io/policy=inherited"
+
+// ClientSettingsPolicy is an Inherited Attached Policy. It provides a way to configure the behavior of the connection
+// between the client and NGINX Gateway Fabric.
+type ClientSettingsPolicy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the desired state of the ClientSettingsPolicy.
+	Spec ClientSettingsPolicySpec `json:"spec"`
+
+	// Status defines the state of the ClientSettingsPolicy.
+	Status ClientSettingsPolicyStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// ClientSettingsPolicyList contains a list of ClientSettingsPolicies.
+type ClientSettingsPolicyList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []ClientSettingsPolicy `json:"items"`
+}
+
+// ClientSettingsPolicySpec defines the desired state of ClientSettingsPolicy.
+type ClientSettingsPolicySpec struct {
+	// TargetRef identifies an API object to apply the policy to.
+	// Object must be in the same namespace as the policy.
+	//
+	// Support: Gateway
```

Comment:
This will need to be updated to include HTTPRoute once we implement that attachment in #1791

Reply:
Alternatively, I can add it now...